certbot/client.py

```diff
@@ -298,7 +298,7 @@ def obtain_certificate(self, domains):
         auth_domains = set(a.body.identifier.value for a in authzr)
         successful_domains = [d for d in domains if d in auth_domains]
 
-        if successful_domains != domains:
+        if self.config.allow_subset_of_names and successful_domains != domains:
```

Comment:
`# This seems redundant, but is necessary because a.body.identifier.value will not be the domain for wildcard certificates` and if that's not true, then a comment explaining what is the case.